pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -156,11 +162,14 @@ library BasePoolMath {
         }
 
         // Calculate the new invariant with fees applied.
-        uint256 invariantWithFeesApplied = computeInvariant(newBalances);
+        // This invariant should be lower than the original one, so we don't need to check invariant ratio bounds again.
+        uint256 invariantWithFeesApplied = pool.computeInvariant(newBalances);
```

Comment:
It seems that we can have a read-only reentrancy attack here. I'll think about cases a little more

Reply:
hmm how so?

Reply:
it seems like I'm a mistake